lib/puppet_x/puppetlabs/scheduled_task/task.rb

```diff
@@ -190,8 +190,21 @@ def self.exists?(job_name)
   #
   def self.delete(task_name)
     task_path = ROOT_FOLDER + task_name
-    task_folder = task_service.GetFolder(folder_path_from_task_path(task_path))
+    folder_path = folder_path_from_task_path(task_path)
+    task_folder = task_service.GetFolder(folder_path)
     task_folder.DeleteTask(task_name_from_task_path(task_path), 0)
+    delete_empty_folder(folder_path)
+  end
+
+  # Delete empty folders
+  def self.delete_empty_folder(folder_path)
+    parent_folder_path = folder_path_from_task_path(folder_path)
```

Comment:
`parent_folder_path` isn't used till later.  may be best to enumerate when it's needed, not before

Reply:
:sob: